website/react-magma-docs/src/pages/api/modal.mdx

```diff
@@ -164,9 +166,38 @@ import {
 export function Example() {
   const [showModal, setShowModal] = React.useState(false);
   const [showModalHeader, setShowModalHeader] = React.useState(false);
+  const [showModalHeaderRef, setShowModalHeaderRef] = React.useState(false);
   const buttonRef = React.useRef();
   const headerButtonRef = React.useRef();
 
+  const [customHeadingRef, setCustomHeadingRef] = React.useState(
+    React.useRef<any>()
+  );
+
+  const handleGetHeaderRef = ref => {
+    setCustomHeadingRef(ref);
+  };
+
+  const onModalShow = () => {
+    setShowModal(true);
+  };
+
+  const onModalClose = () => {
+    setShowModal(false);
+  };
+
+  const onModalShowDefault = () => {
+    setShowDefaultModal(true);
+  };
+
+  const onModalCloseDefault = () => {
+    setShowDefaultModal(false);
+  };
+
+  const onHeadingFocus = () => {
+    customHeadingRef.current.focus();
+  };
+
```

Comment:
This example does most of the logic in line (ex: `onClick={() => setShowModal(false)`). For consistency, should we do the same with the new modal where it makes sense?